src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -722,12 +722,22 @@ private void HandleExternalReplication(DatabaseRecord newRecord, List<IDisposabl
             DropIncomingConnections(changes.RemovedDestiantions, instancesToDispose);
 
             var newDestinations = GetMyNewDestinations(newRecord, changes.AddedDestinations);
-
-            Task.Run(() =>
+            if (newDestinations.Count > 0)
             {
-                // here we might have blocking calls to fetch the tcp info.
-                StartOutgoingConnections(newDestinations, external: true);
-            });
+                Task.Run(() =>
+                {
+                    // here we might have blocking calls to fetch the tcp info.
+                    try
+                    {
+                        StartOutgoingConnections(newDestinations, external: true);
+                    }
+                    catch (Exception e)
```

Comment:
Why will it fail here?

Reply:
It shouldn't but if it does it's better to have it in the logs.